Classes/common/query/CDTQIndexManager.m

```diff
@@ -453,80 +447,29 @@ + (FMDatabaseQueue *)databaseQueueWithDatastore:(CDTDatastore *)datastore
-                 path);
-
-        switch ([FMDatabase isDatabaseUnencryptedAtPath:path]) {
-            case kFMDatabaseUnencryptedNotFound: {
-                LogDebug(@"No index found at %@. Carry on and try to open/create a new not "
-                         @"encrypted index.",
-                         path);
-
-                break;
-            }
-            case kFMDatabaseUnencryptedIsUnencrypted: {
-                LogDebug(@"Index at %@ is not encrypted. No key is neccesary, carry on and try to "
-                         @"open it.",
-                         path);
-
-                break;
-            }
-            case kFMDatabaseUnencryptedIsEncrypted:
-            default: {
-                LogError(@"Index at %@ is encrypted and can not be opened without a key.", path);
-
-                success = NO;
-
-                if (error) {
-                    NSDictionary *userInfo = @{
-                        NSLocalizedDescriptionKey :
-                            NSLocalizedString(@"A key is required to decipher the database", nil)
-                    };
-
-                    *error = [NSError errorWithDomain:CDTQIndexManagerErrorDomain
-                                                 code:CDTQIndexErrorEncryptionKeyError
-                                             userInfo:userInfo];
-                }
-
-                break;
-            }
-        }
-    }
-
-    return success;
-}
-
 + (BOOL)configureDatabase:(FMDatabaseQueue *)database
     withEncryptionKeyProvider:(id<CDTEncryptionKeyProvider>)provider
                         error:(NSError **)error
 {
     __block BOOL success = YES;
 
-    NSString *key = [provider encryptionKey];
-    if (key) {
-        [database inDatabase:^(FMDatabase *db) {
-          success = [db setValidKey:key];
-        }];
+    [database inDatabase:^(FMDatabase *db) {
+      NSError *thisError = nil;
+      success = [db setKeyWithProvider:provider error:&thisError];
+      if (!success) {
+          LogError(@"Problem configuring database with encryption key: %@", thisError);
+      }
+    }];
 
-        if (!success && error) {
-            NSDictionary *userInfo = @{
-                NSLocalizedDescriptionKey :
-                    NSLocalizedString(@"Problem configuring database with encryption key", nil)
-            };
+    if (!success && error) {
+        NSDictionary *userInfo = @{
+            NSLocalizedDescriptionKey :
+                NSLocalizedString(@"Problem configuring database with encryption key", nil)
+        };
 
-            *error = [NSError errorWithDomain:CDTQIndexManagerErrorDomain
-                                         code:CDTQIndexErrorEncryptionKeyError
-                                     userInfo:userInfo];
-        }
+        *error = [NSError errorWithDomain:CDTQIndexManagerErrorDomain
+                                     code:CDTQIndexErrorEncryptionKeyError
+                                 userInfo:userInfo];
```

Comment:
If the configuration does not succeed, the `database` is set to nil [here](https://github.com/cloudant/CDTDatastore/blob/46517-read-db-to-check-if-encrypted/Classes/common/query/CDTQIndexManager.m#L440)